### PR TITLE
BUG: Handle invalid filters consistently

### LINKED ISSUE #282: Make `LogManager.filter_log()` handle invalid filters consistently
## Current Behavior

`LogManager.filter_log()` does not consistently handle invalid filters:

- Filtering an empty log raises `KeyError` because the empty DataFrame has no columns.
- An unknown `field_name` raises Pandas `KeyError`.
- `FilterType` is not validated against the selected field’s actual model type.
- Incompatible field and `FilterType` combinations can leak Pandas `TypeError`.

For example, filtering the textual `user` field as a number causes Pandas to compare a string with an integer:

```python
Filter(
    field_name="user",
    filter_value="10",
    filter_type=FilterType.number,
    operator=">",
)
```

`Filter.parse_filter_value()` converts `"10"` to `10`, but it does not verify that `user` is a numeric field.

## Expected Behavior

- Filtering an empty log returns an empty log of the same model type.
- Unknown fields raise `ValueError` containing the field name.
- `FilterType` is validated against the selected field’s actual model type.
- Incompatible field and filter type combinations raise `ValueError`.

## Suggested Implementation

1. Return an empty log immediately when the source log is empty.
2. Validate that `field_name` exists on the log-entry model.
3. Validate that `filter_type` matches the selected field’s model type. For example, a `str` field such as `user` should use `FilterType.text`, while a `datetime` field such as `timestamp` should use `FilterType.date`.

## Acceptance Criteria

- [ ] Filtering an empty log returns an empty log of the same model type.
- [ ] Filtering an unknown field raises `ValueError` containing the field name.
- [ ] Filtering a text field as a number or date raises `ValueError`.
- [ ] Filtering a datetime field as text or number raises `ValueError`.
- [ ] Filtering a numeric field as text or date raises `ValueError`.
- [ ] No Pandas `KeyError` or `TypeError` escapes for these cases.
- [ ] Existing valid text, number, and date filters continue to work.

### PR DESCRIPTION
**Pull request by nourinmohd**
Resolves #282 

- Added validation in `LogManager.filter_log()` so unknown fields and incompatible `FilterType`/field combinations raise `ValueError` before Pandas filtering.
- Short-circuited empty logs to return an empty log of the same model type.
- Added regression tests for empty logs, unknown fields, and invalid text/date/number filter combinations.

## Checklist

- [x] Tests added (if not, comment why)
- [x] Test coverage equal or up from main (run pytest with `--cov=src/ --cov-report term-missing`)
- [x] If not squash merging, every commit passes tests
- [x] Appropriate [commit prefix](https://upgraded-funicular-eywe4gy.pages.github.io/developing/#commit-prefixes) and precise commit message used
- [ ] All debug prints and unnecessary comments removed
- [x] Docstrings are correct and updated
- [x] Documentation is updated, if necessary
- [x] Latest main rebased/merged into branch
- [x] Added comments on this PR where appropriate to help reviewers
- [x] Moved issue status on project board
- [x] Checked the boxes in this checklist ✅

### CODE REVIEW

File: src/fmu/settings/_resources/log_manager.py
Comment (GibranAlfa) [inline src/fmu/settings/_resources/log_manager.py:61]: Right now invalid filters are accepted when the log is empty because the new validation run only after the empty log early return, so an existing empty log with invalid filter returns `[]` instead of `ValueError`. Suggest to move validation before returning empty log.

Also add/modify test after this to validate empty log are fine for valid filter, but raises when invalid filters.
```suggestion
        if self._cached_dataframe is None:
            log_model: Log[LogEntryType] = self.load()
            if len(log_model) == 0:
                self._cached_dataframe = pd.DataFrame()
            else:
                df_log = pd.DataFrame([entry.model_dump() for entry in log_model])
                self._cached_dataframe = df_log
        df_log = self._cached_dataframe
        self._validate_filter_field(filter)
        if df_log.empty:
            return self.model_class([])
```

File: src/fmu/settings/_resources/log_manager.py [hunk 1]
```diff
@@ -96,3 +104,50 @@ def filter_log(self: Self, filter: Filter) -> Log[LogEntryType]:
...
+        """Return the concrete Pydantic model used for log entries."""
+        model_args = self.model_class.__pydantic_generic_metadata__["args"]
+        return model_args[0]
+
+    @staticmethod
+    def _filter_type_for_annotation(annotation: object) -> FilterType | None:
+        """Return the supported filter type for a model field annotation."""
+        origin = get_origin(annotation)
+        if origin is not None:
+            annotation_args = [
```
Comment (GibranAlfa) [inline src/fmu/settings/_resources/log_manager.py]: suggest to rename this variable to `non_none_args` or `non_none_annotation_args` to be more readable

**Review by GibranAlfa — APPROVED** [review]
nice work 👍
